### PR TITLE
Improve the landscape for README

### PR DESCRIPTION
**Pull request by yaooqinn**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
You can check it here https://github.com/yaooqinn/kyuubi/tree/readme for what it looks like. cc @turboFei @pan3793 @ulysses-you.

**Review by pan3793 — APPROVED** [review]
Awesome!

File: README.md [hunk 1]
```diff
@@ -9,26 +9,70 @@
...
+
+Apache Spark is a unified analytics engine for large-scale data processing. It provides a Distributed SQL Engine, a.k.a, the Spark Thrift Server(STS), designed to be seamlessly compatible with HiveServer2 and get even better performance.
+
+HiveServer2 can identify and authenticate a caller, and then if the caller also has permissions for the YARN queue and HDFS files, it succeeds. Otherwise, it fails. However, on the one hand, STS is a single Spark application, and the user and the queue to which STS belongs are uniquely determined at startup. Consequently, STS cannot leverage cluster managers such as YARN and Kubernetes for resource isolation and sharing or control the access for callers by the single user inside the whole system. On the other hand, the Thrift Server is coupled in the Spark driver's JVM process. This coupled architect puts a high risk on server stability and makes it unable to handle high client concurrency or apply high availability such as load balancing as it is stateful.
```
Comment (turboFei) [inline README.md:46]: >>> However, on the one hand, STS is a single Spark application, and the user and the queue to which STS belongs are uniquely determined at startup.
?
However, on the one hand, STS is a single Spark application, the user and the queue, to which STS belongs, are uniquely determined at startup.

Comment (yaooqinn) [reply README.md:46]: However, on the one hand, STS is a single Spark application. The user and queue to which STS belongs are uniquely determined at startup.

How about？

File: README.md [hunk 2]
```diff
@@ -9,26 +9,70 @@
...
+
+Kyuubi extends the use of STS in a multi-tenant model based on a unified interface and relies on the concept of multi-tenancy to interact with cluster managers to finally gain the ability of resources sharing/isolation and data security. The loosely coupled architecture of the Kyuubi server and engine dramatically improves the client concurrency and service stability of the service itself.
```
Comment (turboFei) [inline README.md:48]: loose?

File: README.md [hunk 3]
```diff
@@ -9,26 +9,70 @@
...
-or batch processing and point queries, or full table scans.
-It embraces Spark and builds an ecosystem on top of it,
-which allows Kyuubi to quickly expand its existing ecosystem and introduce new features,
-such as cloud-native support and `Data Lake/Lake House` support.
-
-The vision of Kyuubi is to unify the portal and become an easy-to-use data lake management platform.
-Different kinds of workloads, such as ETL processing and BI analytics, can be supported by one platform, using one copy of data, with one SQL interface.
+## What is Kyuubi?
+
+Kyuubi is a distributed multi-tenant Thrift JDBC/ODBC server for large-scale data management, processing, and analytics, built on top of Apache Spark and designed to support more engines (i.e., Flink). It has been open-sourced by NetEase since 2018. We are aiming to make Kyuubi an "out-of-the-box" tool for data warehouses and data lakes.
```
Comment (turboFei) [inline README.md:14]: We are aiming to make Kyuubi be an?

File: README.md (hunk 2 above)
Comment (yaooqinn) [reply README.md:48]: loosely is for `coupled` not `architecture`, but need cc: @MrPowers

File: README.md [hunk 4]
```diff
@@ -9,26 +9,70 @@
...
+
+Apache Spark is a unified analytics engine for large-scale data processing. It provides a Distributed SQL Engine, a.k.a, the Spark Thrift Server(STS), designed to be seamlessly compatible with HiveServer2 and get even better performance.
+
+HiveServer2 can identify and authenticate a caller, and then if the caller also has permissions for the YARN queue and HDFS files, it succeeds. Otherwise, it fails. However, on the one hand, STS is a single Spark application. The user and queue to which STS belongs are uniquely determined at startup. Consequently, STS cannot leverage cluster managers such as YARN and Kubernetes for resource isolation and sharing or control the access for callers by the single user inside the whole system. On the other hand, the Thrift Server is coupled in the Spark driver's JVM process. This coupled architect puts a high risk on server stability and makes it unable to handle high client concurrency or apply high availability such as load balancing as it is stateful.
```
Comment (ulysses-you) [inline README.md:46]: `the Thrift Server is coupled in the Spark`  ->  `STS is coupled in the Spark`

File: README.md (hunk 1 above)
Comment (turboFei) [reply README.md]: LGTM

**Comment by yaooqinn** [comment]
Merged this to master for 1.2.0,

**Review by ulysses-you — APPROVED** [review]
it's nice !
